Modules/_lzmamodule.c

```diff
@@ -1393,103 +1408,104 @@ _lzma__decode_filter_properties_impl(PyObject *module, lzma_vli filter_id,
-        ADD_INT_PREFIX_MACRO(m, FILTER_POWERPC) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, MF_HC3) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, MF_HC4) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, MF_BT2) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, MF_BT3) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, MF_BT4) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, MODE_FAST) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, MODE_NORMAL) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, PRESET_DEFAULT) == -1 ||
-        ADD_INT_PREFIX_MACRO(m, PRESET_EXTREME) == -1)
-        return NULL;
+    if (PyModule_AddIntMacro(module, FORMAT_AUTO) == -1 ||
+        PyModule_AddIntMacro(module, FORMAT_XZ) == -1 ||
+        PyModule_AddIntMacro(module, FORMAT_ALONE) == -1 ||
+        PyModule_AddIntMacro(module, FORMAT_RAW) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, CHECK_NONE) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, CHECK_CRC32) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, CHECK_CRC64) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, CHECK_SHA256) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, CHECK_ID_MAX) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, CHECK_UNKNOWN) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_LZMA1) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_LZMA2) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_DELTA) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_X86) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_IA64) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_ARM) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_ARMTHUMB) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_SPARC) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, FILTER_POWERPC) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, MF_HC3) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, MF_HC4) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, MF_BT2) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, MF_BT3) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, MF_BT4) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, MODE_FAST) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, MODE_NORMAL) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, PRESET_DEFAULT) == -1 ||
+        ADD_INT_PREFIX_MACRO(module, PRESET_EXTREME) == -1) {
+        return -1;
+    }
 
-    Error = PyErr_NewExceptionWithDoc(
-            "_lzma.LZMAError", "Call to liblzma failed.", NULL, NULL);
-    if (Error == NULL)
-        return NULL;
+    if (!Error) {
+        Error = PyErr_NewExceptionWithDoc("_lzma.LZMAError", "Call to liblzma failed.", NULL, NULL);
+        if (!Error) {
+            return -1;
+        }
+    }
     Py_INCREF(Error);
-    if (PyModule_AddObject(m, "LZMAError", Error) == -1)
-        return NULL;
+    if (PyModule_AddObject(module, "LZMAError", Error) == -1) {
+        Py_DECREF(Error);
+        return -1;
+    }
 
-    if (PyModule_AddType(m, &Compressor_type) < 0) {
-        return NULL;
+    if (PyModule_AddType(module, &Compressor_type) < 0) {
```

Comment:
`(De)Compresor_type` can be ported to using `PyType_FromSpec` too?

Reply:
@shihai1991 
I 'd like to deal with separate PR as bpo-40077.